primitives/tokens/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Debug, Encode, Decode, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum CurrencyId {
+	Usd,
+	Tranche(u64, u8),
+}
+
+#[macro_export]
+macro_rules! impl_tranche_token {
+	() => {
+		pub struct TrancheToken<T>(core::marker::PhantomData<T>);
+
+		impl<T> pallet_tinlake_investor_pool::TrancheToken<T> for TrancheToken<T>
+		where
+			T: Config,
+			<T as Config>::PoolId: Into<u64>,
+			<T as Config>::TrancheId: Into<u8>,
+			<T as Config>::CurrencyId: From<CurrencyId>,
+		{
```

Comment:
`From<CurrencyId>` refers to type of `CurrencyId` that we import in the context of this trait being used, right? I think we should have hidden imports here, that make it clear, to which `CurrencyId` type the `From` refers to!

Reply:
I'm not sure if we want to keep this here. I would prefer to bundle this as part of pool pallet. thoughts @branan ?